micronota/parsers/tests/test_cmscan.py

```diff
@@ -0,0 +1,187 @@
+            'PASS': '1',
+            'GC_CONTENT': '0.40',
+            'BIAS': '0.0',
+            'BITSCORE': '136.7',
+            'EVALUE': '4e-27',
+            'INC': '!',
+            'DESCRIPTION': '-'
+        }]
+        self.intervals = [
+            [(4977823, 4980727)],
+            [(4831659, 4833190)],
+            [(1915102, 1915281)],
+            [(3794337, 3794487)]
+        ]
+
+        self.seq1 = IntervalMetadata(features={
+            Feature(**self.subsequences[0]): self.intervals[0],
+            Feature(**self.subsequences[1]): self.intervals[1]
+        })
+        self.seq2 = IntervalMetadata(features={
+            Feature(**self.subsequences[2]): self.intervals[2],
+            Feature(**self.subsequences[3]): self.intervals[3]
+        })
+
+
+class ReaderTests(CmscanIOTests):
+    def test_cmscan_to_metadata(self):
+        t = TestCase()
+#       positive controls: will the information from a file parsed to what
+#       we expect
+        assertTrue(self.seq1.__eq__(_cmscan_to_metadata(
+                                                        self.file_valid,
+                                                        rec_num=1)))
+        assertTrue(self.seq2.__eq__(_cmscan_to_metadata(
+                                                        self.file_valid,
+                                                        rec_num=2)))
+
+#       negative control: we are parsing the wrong information from the file,
+#       check if it is really unequal
+        assertTrue(not self.seq1.__eq__(_cmscan_to_metadata(
+                                                            self.file_valid,
+                                                            rec_num=2)))
+
+#       test if parser raises error about an unknown character as strand
+#       identifier
+        t.assertRaisesRegex(CmscanFormatError,
+                            "Unknown strand character",
+                            _cmscan_to_metadata,
+                            self.file_invalidOrientation,
+                            rec_num=1)
+
+#       test if parser complains about non digit characters in positional
+#       arguments
+        t.assertRaisesRegex(CmscanFormatError,
+                            "must be an integer value for the start position "
+                            "of the hit. Here, it is",
+                            _cmscan_to_metadata,
+                            self.file_charInPos,
+                            rec_num=1)
+
+#       test if parser checks for wrong start and stop positions of the hit
+#       in the query sequence
+        t.assertRaisesRegex(CmscanFormatError,
+                            "It might be, that this hit is in fact on the "
+                            "reverse strand. Please check strand orientation "
+                            "and positions",
+                            _cmscan_to_metadata,
+                            self.file_startStopSwop,
+                            rec_num=1)
```

Comment:
Can split these up into multiple test cases, it'll make debugging easier in the future